aten/src/ATen/native/cuda/TensorFactories.cu

```diff
@@ -26,4 +32,39 @@ Tensor& eye_out_cuda(Tensor& result, int64_t n, int64_t m) {
   return result;
 }
 
+namespace {
+template <typename scalar_t>
+void randperm_cuda(Tensor& result, int64_t n, Generator* generator) {
+  result.resize_({n});
+
+  for(int64_t i = 0; i < n; i++) {
```

Comment:
you need to write a CUDA kernel, I dont think writing a host-side function is acceptable here.